<?xml version="1.0"?>
<launch>
  <arg name="machine_name" default="$(optenv HOSTNAME localhost)"/>
  <arg name="config_file" default="$(find system_monitor)/config/system_monitor.yaml"/>
  <arg name="output" default="log"/>

  <group ns="system_monitor">
    <rosparam command="load" file="$(arg config_file)"/>
    <node name="cpu_monitor" pkg="system_monitor" type="cpu_monitor.py" output="$(arg output)" respawn="true"/>
    <node name="hdd_monitor" pkg="system_monitor" type="hdd_monitor.py" output="$(arg output)" respawn="true"/>
    <node name="mem_monitor" pkg="system_monitor" type="mem_monitor.py" output="$(arg output)" respawn="true"/>
    <node name="ntp_monitor" pkg="system_monitor" type="ntp_monitor.py" output="$(arg output)" respawn="true"/>
    <node name="net_monitor" pkg="system_monitor" type="net_monitor.py" output="$(arg output)" respawn="true"/>
    <node name="gpu_monitor" pkg="system_monitor" type="gpu_monitor.py" output="$(arg output)" respawn="true"/>
  </group>

</launch>
